src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -320,10 +331,45 @@ private void FinishBindingOnUIThread(IProgressController controller, IProgressSt
             }
         }
 
+        internal /*for testing purposes*/ void UnpackAdditionalFiles(IProgressController controller, CancellationToken token, IProgressStepExecutionEvents notificationEvents)
+        {
+            if (!this.AdditionalFiles.Any())
+            {
+                return; // Nothing to unpack
+            }
+
+            Solution2 solution = this.projectSystemHelper.GetCurrentActiveSolution();
+            if (solution == null)
+            {
+                VsShellUtils.WriteToGeneralOutputPane(this.owner.ServiceProvider, Strings.FailedToUnpackAdditionalFiles);
+                bool aborted = controller.TryAbort();
+                Debug.Assert(aborted || token.IsCancellationRequested, "Failed to abort the workflow");
+            }
+
+            string solutionRoot = Path.GetDirectoryName(solution.FullName);
+            string root = this.solutionRuleSetWriter.GetOrCreateRuleSetDirectory(PathHelper.ForceDirectoryEnding(solutionRoot));
```

Comment:
PathHelper.ForceDirectoryEnding(solutionRoot) - strange that we need to do it... But the looks of it, is shouldn't [using Path.Combine].
If GetOrCreate... needs it to end with '\' then it needs to call the helper...